### PR TITLE
Add jira branch, expire old failures, track runtimes by directory 

### PR DESCRIPTION
**Pull request by sethpollack**

### CODE REVIEW

File: lib/zspec/tracker.rb [hunk 1]
```diff
@@ -94,6 +104,10 @@ def filter_by_threshold(failure)
       (@sink.time.first - failure["last_failure"].to_i) < @threshold
     end
 
+    def filter_by_expire_threshold(failure)
```
Comment (eabraham) [inline lib/zspec/tracker.rb:107]: Is this a good name for this? This name is a little non intuitive.

File: lib/zspec/presenter.rb [hunk 2]
```diff
@@ -134,6 +144,13 @@ def track_runtimes(results)
       }
     end
 
+    def track_runtimes_by_path(results)
+      key = (results.dig("summary", "file_path") || "").split("/", 4).take(3).join("/")
+      val = results.dig("summary", "duration").to_i
+      @runtimes_by_path[key] = 0 if @runtimes_by_path[key].nil?
```
Comment (eabraham) [inline lib/zspec/presenter.rb:150]:
```suggestion
      @runtimes_by_path[key] ||= 0
```

File: lib/zspec/presenter.rb [hunk 3]
```diff
@@ -134,6 +144,13 @@ def track_runtimes(results)
       }
     end
 
+    def track_runtimes_by_path(results)
+      key = (results.dig("summary", "file_path") || "").split("/", 4).take(3).join("/")
```
Comment (eabraham) [inline lib/zspec/presenter.rb:148]:
```suggestion
      key = results.dig("summary", "file_path")&.split("/", 4)&.take(3)&.join("/") || ""
```
Would this be clearer?

File: lib/zspec/tracker.rb (hunk 1 above)
Comment (sethpollack) [reply lib/zspec/tracker.rb:107]: any suggestions?

Comment (eabraham) [reply lib/zspec/tracker.rb:107]: Is `filter_by_expire_threshold ` always `!filter_by_threshold`? If so can we name it inclusive/exclusive threshold filter?

Comment (sethpollack) [reply lib/zspec/tracker.rb:107]: Ok

**Review by eabraham — APPROVED** [review]

**Pull request merged**